mne/export/tests/test_export.py

```diff
@@ -46,6 +51,96 @@ def test_export_raw_eeglab(tmpdir):
     assert_allclose(raw.get_data(), raw_read.get_data())
 
 
+@pytest.mark.skipif(not _check_edflib_installed(strict=False),
+                    reason='edflib-python not installed')
+@pytest.mark.parametrize('format', ['edf', 'bdf'])
+def test_integer_sfreq_edf_and_bdf(tmp_path, format):
+    """Test saving a Raw array with integer sfreq to EDF and BDF."""
+    info = create_info(['1', '2', '3'], sfreq=1000, ch_types='eeg')
+    data = np.random.random(size=(3, 1000))
+    raw = RawArray(data, info)
+
+    temp_fname = op.join(str(tmp_path), f'test.{format}')
+
+    raw.export(temp_fname)
+    if format == 'edf':
+        raw_read = read_raw_edf(temp_fname, preload=True)
+    elif format == 'bdf':
+        raw_read = read_raw_bdf(temp_fname, preload=True)
+
+    assert raw.ch_names == raw_read.ch_names
+    # only compare the original length, since extra zeros are appended
+    orig_raw_len = len(raw)
+    assert_array_almost_equal(
+        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=3)
+    assert_allclose(
+        raw.times, raw_read.times[:orig_raw_len], rtol=0, atol=1e-5)
+
+
+@pytest.mark.skipif(not _check_edflib_installed(strict=False),
+                    reason='edflib-python not installed')
+@pytest.mark.parametrize(
+    ['dataset', 'format'], [
+        ['test', 'edf'],
+        ['test', 'bdf'],
+        ['misc', 'edf'],
+        ['misc', 'bdf']
+    ])
+def test_export_raw_edf_and_bdf(tmp_path, dataset, format):
+    """Test saving a Raw instance to EDF and BDF format."""
+    if dataset == 'test':
+        try:
+            import importlib_resources as imp_resrc  # py 3.7
+        except ImportError:
+            import importlib.resources as imp_resrc  # py 3.8+
+        import mne.io.tests.data
+        with imp_resrc.path(mne.io.tests.data, 'test_raw.fif') as fname:
+            raw = read_raw_fif(fname)
```

Comment:
I would take the fname by relative path udpates starting from __file__

it's how we do it else where. See for example:

https://github.com/mne-tools/mne-python/blob/main/mne/tests/test_evoked.py#L26

Reply:
I requested this change, which is possible now that py3.7 is our min version. I'd like to see all of them done this way, but if you object @agramfort then I guess just revert this @adam2392

Reply:
@agramfort said "fine with him", so I'll keep it

Ref: https://github.com/mne-tools/mne-python/pull/9643#issuecomment-903116275